src/plots/plots.js

```diff
@@ -1178,6 +1181,14 @@ plots.supplyTraceDefaults = function(traceIn, traceOut, colorIndex, layout, trac
 
         plots.supplyTransformDefaults(traceIn, traceOut, layout);
     }
+    else if(_module && Registry.traceIs(traceOut, 'alwaysSupplyDefaults')) {
+        // Some types need *something* from supplyDefaults always, even if
+        // visible: false. Looking at you scatter: stack options even from
+        // hidden traces can control other traces in the stack.
+        // These types should bail out ASAP if visible is false.
+        // But we don't need any other cross-module attrs ^^ in this case.
+        _module.supplyDefaults(traceIn, traceOut, defaultColor, layout);
```

Comment:
I don't think so - this `else if` is in reference to an `if(visible)` above. That change in #2860 ensures we still call `plot` for these traces and base plot modules but `supplyDefaults` still bails.

Reply:
Ok, my mistake I meant to write _gets called even on traces with `visible:'legendonly'`_ which of course predates #2860 (my shameless reference to a PR I made did go so well here :face_with_head_bandage: 

So, w/o this block, stack options even from hidden traces (i.e. legendonly traces) can _still_ control other traces in the stack. 

In other words, do we need to extra piece of logic "just for" `visible:false`?

Reply:
>  do we need to extra piece of logic "just for" `visible:false`?

Hmm, good question. I could see arguments either way. On the one hand you could say `visible: false` should be the equivalent of removing the trace altogether, just keeping it available in case you want it later. On the other hand I could imagine someone making an app with its own legend or equivalent outside of plotly.js, and being confused that when they hide the bottom stack item the stack config is lost...

Reply:
Alright, you've won me over, both have some arguments in their favor but ignoring `visible: false` gives cleaner, nicer code and is probably more consistent with the rest of our logic. https://github.com/plotly/plotly.js/pull/2960/commits/9fdf5226430c756a8844847e1f2471e76e60a72c